pkg/encoding/encoding_test.go

```diff
@@ -1,185 +1,866 @@
-	{"8859_1_pt.html", "ISO-8859-1", false, false},
-	{"ascii.txt", "ISO-8859-1", false, false},
-	{"big5.html", "Big5", false, false},
-	{"candide-gb18030.txt", "windows-1252", false, false}, // should be GB18030
-	{"candide-utf-16le.txt", "ISO-8859-1", false, false},  // should be UTF-16LE
-	{"candide-utf-32be.txt", "UTF-32BE", false, false},
-	{"candide-utf-8.txt", "UTF-8", false, false},
-	{"candide-windows-1252.txt", "ISO-8859-1", false, false}, // should be windows-1252
-	{"cp865.txt", "windows-1252", false, false},
-	{"euc_jp.html", "EUC-JP", false, false},
-	{"euc_kr.html", "EUC-KR", false, false},
-	{"gb18030.html", "GB18030", false, false},
-	{"html.html", "ISO-8859-1", false, false},
-	{"html.iso88591.html", "ISO-8859-1", false, false},
-	{"html.svg.html", "ISO-8859-1", false, false},
-	{"html.usascii.html", "ISO-8859-1", false, false},
-	{"html.utf8bomdetect.html", "UTF-8", false, false},
-	{"html.utf8bom.html", "UTF-8", false, false},
-	{"html.utf8bomws.html", "UTF-8", false, false},
-	{"html.utf8.html", "ISO-8859-1", false, false}, // should be UTF-8 ?
-	{"html.withbr.html", "ISO-8859-1", false, false},
-	{"iso88591.txt", "ISO-8859-1", false, false},
-	{"koi8_r.txt", "ISO-8859-1", false, false}, // should be KOI8-R
-	{"latin1.txt", "ISO-8859-1", false, false},
-	{"rashomon-euc-jp.txt", "EUC-JP", false, false},
-	{"rashomon-iso-2022-jp.txt", "ISO-2022-JP", false, false},
-	{"rashomon-shift-jis.txt", "Shift_JIS", false, false},
-	{"rashomon-utf-8.txt", "UTF-8", false, false},
-	{"shift_jis.html", "Shift_JIS", false, false},
-	{"sunzi-bingfa-gb-levels-1-and-2-hz-gb2312.txt", "Big5", false, false}, // should be GB18030
-	{"sunzi-bingfa-gb-levels-1-and-2-utf-8.txt", "UTF-8", false, false},
-	{"sunzi-bingfa-simplified-gbk.txt", "GB18030", false, false},
-	{"sunzi-bingfa-simplified-utf-8.txt", "UTF-8", false, false},
-	{"sunzi-bingfa-traditional-big5.txt", "Big5", false, false},
-	{"sunzi-bingfa-traditional-utf-8.txt", "UTF-8", false, false},
-	{"unsu-joh-eun-nal-euc-kr.txt", "EUC-KR", false, false},
-	{"unsu-joh-eun-nal-utf-8.txt", "UTF-8", false, false},
-	{"utf16bebom.txt", "UTF-16BE", false, false},
-	{"utf16lebom.txt", "UTF-16LE", false, false},
-	{"utf16.txt", "UTF-16LE", false, false},
-	{"utf32bebom.txt", "UTF-32BE", false, false},
-	{"utf32lebom.txt", "UTF-32LE", false, false},
-	{"utf8_bom.html", "UTF-8", false, false},
-	{"utf8.html", "UTF-8", false, false},
-	{"utf8-sdl.txt", "windows-1252", false, false}, // should be UTF-8
-	{"utf8.txt", "UTF-8", false, false},
-	{"utf8.txt-encoding-test-files.txt", "UTF-8", false, false},
-	// Issues
-	{"issue252.txt", "UTF-8", false, false},
-}
-
-func TestDecodeBytesText(t *testing.T) {
-	for _, tt := range decodeTextTests {
-		filename := "testdata/" + tt.filename
-		rawFileContent, err := os.ReadFile(filename)
+const minConfidenceToFailTests = 1
+
+const minConfidenceToSetEncoding = 10
+
+const defaultConfidence = 1
+
+const testResultsJson = "test-results.json"
```

Comment:
do I understand this correctly as being a test snapshot file? If so, why not reuse snaps - there the order of the tests would not matter?

Reply:
To be honest, I haven't worked with go-snaps before. Are you suggesting we should here?

Note: If `EDITORCONFIG_ADD_NEW_FILES=1` is set in the environment, the tests suite will scan for new files in `testdata` and add them to the `test-results.json`. Otherwise the suite runs only on the `testdata` files listed in the file.

I found the file very helping in my debugging, as I could run a `git diff` after a run to see if anything changed. That was a lot easier than scanning the log output for hundreds of files.

Reply:
okay, so basically you implement the reverse behaviour of snaps. Snaps will add new snapshots, but never change existing snapshots unless `UPDATE_SNAPS` is set to a true value.

In this use case I would think using `snaps.MatchStandaloneJSON(t, someValue)` would be better.
https://github.com/gkampitakis/go-snaps?tab=readme-ov-file#matchjson

This would mean:
- always scan for testfiles
- during teardown call `snaps.MatchStandaloneJSON()` for each test. Although one might argue that matching the snapshot inline would be easier, but right now the test architecture touches the central `tests` slice multiple times, if I understand it correctly.

snaps itself will then generate test failures when a previously created snapshot is not matched.